server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -91,63 +115,224 @@ private void registerListeners() {
         clusterService.addListener(this);
     }
 
+    // When the monitoring interval changed we reschedule the health monitoring task with the new interval.
     void setMonitorInterval(TimeValue monitorInterval) {
         this.monitorInterval = monitorInterval;
-        maybeScheduleNow();
+        cancelExistingSchedule();
+        maybeStartNewSchedule();
     }
 
+    // When the health node is enabled we try to schedule a monitoring task if it is not
+    // already scheduled, no need to reschedule if it is already there since there was
+    // no configuration change. When the health node is disabled we cancel the schedule.
     void setEnabled(boolean enabled) {
         this.enabled = enabled;
-        maybeScheduleNow();
+        if (enabled && isScheduled() == false) {
+            maybeStartNewSchedule();
+        }
+        if (enabled == false) {
+            cancelExistingSchedule();
+        }
     }
 
-    /**
-     * We always check if the prerequisites are fulfilled and if the health node
-     * is enabled before we schedule a monitoring task.
-     */
-    private void maybeScheduleNextRun(TimeValue time) {
+    private void cancelExistingSchedule() {
+        // If there is an existing schedule, cancel it
+        Scheduler.Cancellable existingTask = scheduledTask.get();
+        if (existingTask != null) {
+            existingTask.cancel();
+        }
+    }
+
+    private void maybeStartNewSchedule() {
         if (prerequisitesFulfilled && enabled) {
-            threadPool.scheduleUnlessShuttingDown(time, ThreadPool.Names.MANAGEMENT, this::monitorHealth);
+            // We synchronize on this block to ensure that we will not override an existing
+            // schedule that is not cancelled
+            // TODO check in which thread are the setting updates executed
+            synchronized (this) {
```

Comment:
I don't _think_ this needs to be synchronized, does it? This can only be called from the object initialization code or the cluster change event thread (which only begins calling this object at the very end of object initialization).

Reply:
You are right. `synchronized` removed!